driver-core/src/test/java/com/datastax/driver/core/HostResolutionReconnectionTest.java

```diff
@@ -0,0 +1,86 @@
+package com.datastax.driver.core;
+
+import java.net.InetSocketAddress;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.burningwave.tools.net.DefaultHostResolver;
+import org.burningwave.tools.net.HostResolutionRequestInterceptor;
+import org.burningwave.tools.net.MappedHostResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.Test;
+
+public class HostResolutionReconnectionTest {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(HostResolutionReconnectionTest.class);
+
+  @Test(groups = "isolated")
+  public void should_reconnect_to_different_cluster() {
```

Comment:
Let's keep test names in sync with `scylla-4.x`  branch:
```suggestion
  public void replace_cluster_test() {
```